### PR TITLE
Add shuffle function

### PR DESCRIPTION
**Pull request by 1v4n4**
Implemented [Fisher-Yates algorithm](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle) to randomize targets

### CODE REVIEW

File: targets.js [hunk 1]
```diff
@@ -76,4 +76,16 @@ const filter = (targets, criteria) => {
   throw new Error ("don't know how to filter ",criteria);
 }
 
-module.exports = {getTargets, filter };
+const shuffle = (targets) => {
+
+  const shuffled = [...targets];
```
Comment (tttp) [inline targets.js]: why are you working on a copy? the result is going to be assigned to the source object anyway

File: targets.js [hunk 2]
```diff
@@ -76,4 +76,16 @@ const filter = (targets, criteria) => {
   throw new Error ("don't know how to filter ",criteria);
 }
 
-module.exports = {getTargets, filter };
+const shuffle = (targets) => {
```
Comment (tttp) [inline targets.js]: could you put it into util.js? we might use it elsewhere too

File: prepare.js [hunk 3]
```diff
@@ -18,15 +18,15 @@ const {
...
 const { preview, initPreview } = require("./mailer");
 const { getStats } = require("./server");
 let csv = "name,email,saluation,gender,language,area,external_id";
 
 const argv = require("minimist")(process.argv.slice(2), {
-  boolean: ["help", "dry-run", "verbose", "csv"],
+  boolean: ["help", "dry-run", "verbose", "csv", "shuffle"],
   string: ["file", "lang", "template", "fallback", "preview"],
   //  unknown: (param) => {param[0] === '-' ? console.error("invalid parameter",param) || process.exit(1) : true},
   default: { template: "default", csv: true, min: 0 },
```
Comment (tttp) [inline prepare.js]: put shuffle:true in the default

**Review by tttp — CHANGES_REQUESTED** [review]

File: targets.js (hunk 1 above)
Comment (1v4n4) [reply targets.js]: Done, works the same

File: prepare.js (hunk 3 above)
Comment (1v4n4) [reply prepare.js]: done

File: targets.js (hunk 2 above)
Comment (1v4n4) [reply targets.js]: done

**Comment by 1v4n4** [comment]
@tttp All changed and tested

**Pull request merged**